.github/workflows/surge-preview.yml

```
@@ -0,0 +1,38 @@
+
+jobs:
+  preview:
+    runs-on: ubuntu-latest
+    env:
+      PREVIEW: true
+    steps:
+      - uses: actions/checkout@9c091bb21b7c1c1d1991bb908d89e4e9dddfe3e0
+        with:
+          persist-credentials: false
+      - name: Build preview
+        if: ${{ secrets.SURGE_TOKEN != '' }}
```

actionlint (1.7.12)

[error] context "secrets" is not allowed here. available contexts are "env", "github", "inputs", "job", "matrix", "needs", "runner", "steps", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)